### PR TITLE
fix #237 use standard chain naming conventions

### LINKED ISSUE #237: We should switch to more standard naming conventions when working with chain names
**The issue**
Currently, across our SDKs and libraries, we reference the following chain specifiers:

- ethereum
- optimism
- arbitrum
- polygon
- ethereum-goerli
- optimism-goerli
- arbitrum-goerli
- polygon-mumbai
- optimism-goerli-staging
- localhost

However, these specifiers are specific to Tableland. For instance, `ethersjs` (and https://chainlist.org) use the [following equivalent set](https://github.com/ethers-io/ethers.js/blob/master/packages/networks/src.ts/index.ts#L135):

- mainnet (alias homestead)*
- optimism
- arbitrum
- matic*
- goerli*
- optimism-goerli
- arbitrum-goerli
- maticmum*
- optimism-goerli (with our own `staging` suffix?)
- localhost

For localhost, I think hardhat has a convention here, and I think perhaps @joewagner also had an opinion on this one?

**Describe the solution you'd like**
It would be useful to adopt the above naming conventions in this repo, and in our SDK and CLI tooling, so that a) it is easier for devs to cross over when dealing with chains and networks, and b) so that our code is more directly compatible with ethers and other projects.

**Describe alternatives you've considered**
At the moment, supporting our own naming conventions require a mapping in the SDK or CLI from Tableland names to external names. This isn't the end of the world, but it does mean that we end up duplicating names across places, rather than having a single source of truth that we can just use "as is".

### PR DESCRIPTION
**Pull request by joewagner**

### CODE REVIEW

File: network.ts [hunk 1]
```diff
@@ -1,43 +1,55 @@
...
+  matic: "0x5c4e6A9e5C1e1BF445A062006faF19EA6c49aFeA",
+  goerli: "0xDA8EA22d092307874f30A1F277D1388dca0BA97a",
   "optimism-goerli": "0xC72E8a7Be04f2469f8C2dB3F1BdF69A7D516aBbA",
   "arbitrum-goerli": "0x033f69e8d119205089Ab15D340F5b797732f646b",
-  "polygon-mumbai": "0x4b48841d4b32C4650E4ABc117A03FE8B51f38F68",
+  maticmum: "0x4b48841d4b32C4650E4ABc117A03FE8B51f38F68",
   "optimism-goerli-staging": "0xfe79824f6E5894a3DD86908e637B7B4AF57eEE28",
-  localhost: "",
+  localhost: localTablelandAddr,
+  "local-tableland": localTablelandAddr,
```
Comment (joewagner) [inline network.ts:34]: There was a few different conversations today, but I __think__ we landed on supporting both of these as a aliases for a hardhat node running with a local validator connected.

File: .prettierignore [hunk 2]
```diff
@@ -4,5 +4,5 @@ artifacts
 cache
 coverage*
 gasReporterOutput.json
-network.d.ts
+*.d.ts
```
Comment (joewagner) [inline .prettierignore:7]: prettier was fixing ignored files, this saves the it from doing the unnecessary work.

**Comment by joewagner** [comment]
@asutula @carsonfarmer Here are the updates to the naming conventions, definitely check that I got this right.  
We can't publish this as v3 without breaking semver, so these changes will only affect the upcoming ship and sdk refactor.  In the SDK refactor we will have the opportunity to use the `baseURIs` exported here to discover the gateway url.

**Comment by carsonfarmer** [comment]
This looks great. Does this also require any updates to deploy scripts etc, now that the target chain names have changed?

**Review by carsonfarmer — DISMISSED** [review]
👍 thanks!

**Comment by joewagner** [comment]
> Does this also require any updates to deploy scripts etc, now that the target chain names have changed?

Yes, definitely. I missed that, but just pushed change that should make that correct.

**Review by carsonfarmer — APPROVED** [review]
Yeh nice, thanks for double checking.

**Pull request merged**